### PR TITLE
clear fluid api: warpctc, nce, identity_loss

### PR DESCRIPTION
**Pull request by isky-cd**
### PR types

Others
### PR changes

APIs
### Describe

clear fluid api: warpctc, nce, identity_loss

### CODE REVIEW

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -4044,3 +4046,182 @@ def soft_margin_loss(input, label, reduction='mean', name=None):
...
+
+            # lengths of logit sequences
+            seq_lens = [2,6]
+            # lengths of label sequences
+            label_lens = [2,3]
+            # class num
+            class_num = 5
+
+            paddle.enable_static()
+            logits = fluid.data(name='logits',shape=[None, class_num+1],
```
Comment (sneaxiy) [inline python/paddle/nn/functional/loss.py]: No fluid codes.

File: python/paddle/incubate/nn/loss.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        Variable: The loss ``Tensor`` with the specified reduction applied.
+
+    Examples:
+
+        .. code-block:: python
+
+            import paddle.fluid as fluid
+            import paddle
+            paddle.enable_static()
+            loss = fluid.data(name="loss", shape=[-1, 1], dtype="float32")
```
Comment (sneaxiy) [inline python/paddle/incubate/nn/loss.py]: No fluid code

File: python/paddle/nn/functional/loss.py (hunk 1 above)
Comment (isky-cd) [reply python/paddle/nn/functional/loss.py]: 这个建业说之后他们集中改

File: python/paddle/incubate/nn/loss.py (hunk 2 above)
Comment (isky-cd) [reply python/paddle/incubate/nn/loss.py]: 这个建业说之后他们集中改

**Review by sneaxiy — DISMISSED** [review]

**Review by jzhang533 — COMMENTED** [review]
新引入API，需经过评审：http://agroup.baidu.com/paddlepaddle/view/office/4139324

File: python/paddle/incubate/nn/loss.py (hunk 2 above)
Comment (zoooo0820) [reply python/paddle/incubate/nn/loss.py]: 示例代码的部分，有替代调用路径的，还是需要相应替换为2.0的使用方法 （static下的data， 文件开头import 的部分可以看看framework, common_ops_import)

Comment (isky-cd) [reply python/paddle/incubate/nn/loss.py]: 已修改

**Comment by isky-cd** [comment]
> 新引入API，需经过评审：http://agroup.baidu.com/paddlepaddle/view/office/4139324


目前nce 和indentity_loss已经是公开API，只是将实现迁移了过来。warpctc由于ctc_loss需要调用，目前已经删除文档，不推荐用户使用。

**Review by jzhang533 — DISMISSED** [review]
LGTM

**Review by sneaxiy — DISMISSED** [review]

**Review by jzhang533 — DISMISSED** [review]

File: python/paddle/incubate/nn/loss.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+from paddle.fluid.layer_helper import LayerHelper
+from paddle.framework import (
+    _non_static_mode,
+)
+from paddle.fluid.data_feeder import check_variable_and_dtype
+from paddle import _legacy_C_ops
+
+
+def identity_loss(x, reduction="none"):
```
Comment (XiaoguangHu01) [inline python/paddle/incubate/nn/loss.py:23]: 这是一个新增api吗？为啥是放在incubate的下呢？
新增api需要走一下正式的评审流程，然后放到合适的位置。

File: python/paddle/nn/functional/loss.py [hunk 4]
```diff
@@ -4016,3 +4018,65 @@ def soft_margin_loss(input, label, reduction='mean', name=None):
         return paddle.mean(out, name=name)
     else:
         return out
+
+
+def warpctc(
```
Comment (XiaoguangHu01) [inline python/paddle/nn/functional/loss.py]: 这里为什么需要增加warpctc的接口呢？
是否可以调用ctc_loss来实现？
https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/ctc_loss_cn.html
不应该增加一个相似的接口

File: python/paddle/incubate/nn/loss.py (hunk 3 above)
Comment (isky-cd) [reply python/paddle/incubate/nn/loss.py:23]: 这个是已经是公开API，现在只是将实现搬过来，还有identity_loss原本就是在incubate下调用的

File: python/paddle/nn/functional/loss.py (hunk 4 above)
Comment (isky-cd) [reply python/paddle/nn/functional/loss.py]: ctc_loss是的实现需要依赖warpctc，目前已经删除warpctc相关文档，并不推荐用户使用。

File: python/paddle/fluid/tests/unittests/test_layers.py [hunk 5]
```diff
@@ -4504,7 +4504,7 @@ def test_warpctc_with_padding(self):
             predict = layers.data(
                 name='predict', shape=[4, 4, 8], dtype='float32'
             )
-            output = layers.warpctc(
+            output = paddle.nn.functional.loss.warpctc(
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/tests/unittests/test_layers.py]: 那这里应该调用ctc_loss，而不是warpctc内部接口

File: python/paddle/incubate/nn/loss.py (hunk 3 above)
Comment (XiaoguangHu01) [reply python/paddle/incubate/nn/loss.py:23]: ok

File: python/paddle/nn/functional/loss.py (hunk 4 above)
Comment (XiaoguangHu01) [reply python/paddle/nn/functional/loss.py]: ok

**Review by sneaxiy — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
